pkg/testutils/bazel.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package testutils
+
+import (
+	"os"
+	"path"
+	"strings"
+	"testing"
+)
+
+const testSrcDirEnv = "TEST_SRCDIR"
+const testTmpDirEnv = "TEST_TMPDIR"
+const testWorkspaceEnv = "TEST_WORKSPACE"
+const testTargetEnv = "TEST_TARGET"
+
+// RunningUnderBazel returns true if the test is executed by bazel.
+func RunningUnderBazel() bool {
+	return os.Getenv(testSrcDirEnv) != ""
+}
+
+func requireEnv(env string) string {
+	if v := os.Getenv(env); v != "" {
+		return v
+	}
+	panic("expected value for env " + env)
+}
+
+// TestSrcDir returns the path to the "source" tree.
+//
+// If running under bazel, this will point to a private, *readonly*
+// directory containing symlinks (or copies) of the test data depencies.
+// This directory must be treated readonly.  It's an error to try to modify
+// anything under this directory: though the operation may succeed, the test
+// would not be hermetic, and may fail under other environments.
+func TestSrcDir() string {
+	// If testSrcDirEnv is not set, it means we are not running under bazel,
+	// and so we can use "" as our directory which should point to the
+	// src root.
+	if srcDir := os.Getenv(testSrcDirEnv); srcDir != "" {
+		return srcDir
+	}
+	return ""
+}
+
+// TestTempDir returns a temporary directory and a cleanup function for a test.
+func TestTempDir(t testing.TB) (string, func()) {
+	if RunningUnderBazel() {
+		// Bazel sets up private temp directories for each test.
+		return requireEnv(testTmpDirEnv), func() {}
+	}
+	return TempDir(t)
+}
+
+// bazeRelativeTargetPath returns relative path to the package
+// of the current test.
+func bazelRelativeTargetPath() []string {
+	target := os.Getenv(testTargetEnv)
+	if target == "" {
+		return nil
+	}
+
+	// Strip out leading package "//"
+	target = strings.TrimPrefix(target, "//")
+
+	// Strip out ":XXX" from the last component
```

Comment:
Why not split on ":", drop the last name, and then split on "/" to get the relative target path? 

It would also make it easier to add a comment up top showing the transformation taking place here. If I've understood correctly, it's the following: `//pkg/ccl/importccl:importccl_test -> pkg/ccl/importccl` (?). 

I was actually a bit surprised that this was not something provided out of the box, or at least easier. I assume we've looked at https://docs.bazel.build/versions/master/test-encyclopedia.html#location and found it inapplicable?

> //foo/bar:unittest is the directory $(WORKSPACE)/$(BINDIR)/foo/bar/unittest.runfiles

Reply:
Somewhat related, I couldn't find official documentation on it, but https://github.com/bazelbuild/bazel/blob/c266ac966761c4b3d8a408a03e407505c93effdd/tools/bash/runfiles/runfiles.bash#L57 has a reference to `RUNFILES_DIR`. I'm curious if that would help. Feel free to ignore though.